From 5e640e96e1224868456fda4cc2c9e8cca273a79c Mon Sep 17 00:00:00 2001
From: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
Date: Mon, 17 May 2021 22:50:25 +0200
Subject: [PATCH 29/96] dt-bindings: firmware: Document the Amlogic
 Meson6/8/8b/8m2 TrustZone

Amlogic Meson6/8/8b/8m2 SoCs can optionally use a TrustZone secure
firmware. This prevents anything outside of the TEE (Trusted
Execution Environment aka TrustZone secure firmware) from accessing
certain functionality of these SoCs, such as (but not limited to):
Bringing up/down secondary SMP cores, accessing the eFuse and getting
the SoC misc version.
ARM SMCCC is used for communication with the TrustZone secure
firmware.

Signed-off-by: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
---
 .../amlogic,meson-mx-trustzone-firmware.yaml  | 47 +++++++++++++++++++
 1 file changed, 47 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/firmware/meson/amlogic,meson-mx-trustzone-firmware.yaml

diff --git a/Documentation/devicetree/bindings/firmware/meson/amlogic,meson-mx-trustzone-firmware.yaml b/Documentation/devicetree/bindings/firmware/meson/amlogic,meson-mx-trustzone-firmware.yaml
new file mode 100644
index 000000000..1e0e19a35
--- /dev/null
+++ b/Documentation/devicetree/bindings/firmware/meson/amlogic,meson-mx-trustzone-firmware.yaml
@@ -0,0 +1,47 @@
+# SPDX-License-Identifier: (GPL-2.0 OR BSD-2-Clause)
+# Copyright 2021 Martin Blumenstingl <martin.blumenstingl@googlemail.com>
+%YAML 1.2
+---
+$id: "http://devicetree.org/schemas/firmware/meson/amlogic,meson-mx-trustzone-firmware.yaml#"
+$schema: "http://devicetree.org/meta-schemas/core.yaml#"
+
+title: Amlogic Meson6/8/8b/8m2 TrustZone secure firmware
+
+description: |
+  Amlogic Meson6/8/8b/8m2 SoCs can optionally use a TrustZone secure
+  firmware. This prevents anything outside of the TEE (Trusted
+  Execution Environment aka TrustZone secure firmware) from accessing
+  certain functionality of these SoCs, such as (but not limited to):
+  Bringing up/down secondary SMP cores, accessing the eFuse and getting
+  the SoC misc version.
+  ARM SMCCC is used for communication with the TrustZone secure
+  firmware.
+
+maintainers:
+  - Martin Blumenstingl <martin.blumenstingl@googlemail.com>
+
+properties:
+  compatible:
+    oneOf:
+      - items:
+          - enum:
+              - amlogic,meson6-trustzone-firmware
+              - amlogic,meson8-trustzone-firmware
+              - amlogic,meson8b-trustzone-firmware
+              - amlogic,meson8m2-trustzone-firmware
+          - const: amlogic,meson-mx-trustzone-firmware
+
+required:
+  - compatible
+
+additionalProperties: false
+
+examples:
+  - |
+    firmware {
+      trustzone-firmware {
+        compatible = "amlogic,meson8m2-trustzone-firmware",
+                     "amlogic,meson-mx-trustzone-firmware";
+      };
+    };
+...
-- 
2.45.1

